### PR TITLE
tcsh: Fix autocompletion for no single-hyphen args

### PR DESCRIPTION
**Pull request by simaoafonso-pwt**
If the parser has no single-hyphen options, the tcsh template would
generate a syntax error.

Include a hack to that the autocompletion for the double-hyphen
selection does not add an extra space at the end.

---

Another fix to the tcsh template support, see #56 #57 #58

### CODE REVIEW

File: shtab/__init__.py [hunk 1]
```diff
@@ -714,27 +716,33 @@ def recurse_parser(cparser, positional_idx, requirements=None):
...
-        'c/-/(${optionals_single_str} -)/' \\
+        'c/-/(${optionals_single_str})/' \\
         ${optionals_special_str} \\
         'p/*/()/'""").safe_substitute(
         preamble=("\n# Custom Preamble\n" + preamble +
                   "\n# End Custom Preamble\n" if preamble else ""), root_prefix=root_prefix,
-        prog=parser.prog, optionals_double_str=' '.join(optionals_double),
-        optionals_single_str=' '.join(optionals_single),
+        prog=parser.prog, optionals_double_str=' '.join(sorted(optionals_double)),
+        optionals_single_str=' '.join(sorted(optionals_single)),
```
Comment (casperdcl) [inline shtab/__init__.py:745]: why sort these? Python's [dicts are insertion ordered](https://peps.python.org/pep-0468/), meaning you can call `.add_argument()` in the order you want things to appear.

Comment (simaoafonso-pwt) [reply shtab/__init__.py:745]: It was only so that the generated file was better looking. I'll remove those.

**Review by casperdcl — CHANGES_REQUESTED** [review]
this seems to break things a little.

before:

```sh
main$ eval "$(PYTHONPATH=. python examples/customcomplete.py completion bash)"
main$ customcomplete <TAB>
completion  process
```

after:

```sh
fix-singlehyphen$ customcomplete <TAB>
fix-singlehyphen$ customcomplete completion
```

**Comment by simaoafonso-pwt** [comment]
> This seems to break things a little (at least on `bash`).

That's weird, I only changed the csh function, and this commit to avoid test failures.

b05c0c76dd114deb103484e0d4dab2e6f312d9bf

I should rebase and retest this.

**Comment by simaoafonso-pwt** [comment]
Rebased and re-tested, including that test you mentioned, works for me.

**Comment by casperdcl** [comment]
Thanks! Rebased now over #139 (should merge that first). Better now?

**Comment by simaoafonso-pwt** [comment]
LGTM, you can merge this.

**Review by casperdcl — APPROVED** [review]
lgtm; thanks!

I removed the `examples/doublecomplete.py` since it doesn't seem required.

Maybe worth adding a follow-up test (in `tests/test_shtab.py`) instead.

**Pull request merged**